### ISSUE TITLE
Does not contain a valid DateToken issue happen During  React native native upgrade to 73.4

### ISSUE THREAD

**Issue by velrajpaulraj** [BeTomorrow/react-native-file-logger#71]
getting this issue  while upgrade  from 63 to 73.4. Does anybody is getting this error.Please suggest the solution.Thank you.

FATAL EXCEPTION: mqt_native_modules
                                                                                                    Process: com.tonik.mobile, PID: 9633
                                                                                                    java.lang.IllegalStateException: FileNamePattern [/storage/emulated/0/Android/data/com.tonik.mobile/cache/logs/com.tonik.mobile-%i.log] does not contain a valid DateToken
                                                                                                    	at com.morfeus.android.mfsdk.s25.start(SourceFile:1)
                                                                                                    	at com.morfeus.android.mfsdk.up4.start(SourceFile:1)
                                                                                                    	at com.morfeus.android.mfsdk.t25.start(SourceFile:1)
                                                                                                    	at com.morfeus.android.mfsdk.vp4.start(SourceFile:1)
                                                                                                    	at com.betomorrow.rnfilelogger.FileLoggerModule.configure(SourceFile:170)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.facebook.react.bridge.JavaMethodWrapper.invoke(SourceFile:143)
                                                                                                    	at com.facebook.react.bridge.JavaModuleWrapper.invoke(SourceFile:22)
                                                                                                    	at com.facebook.jni.NativeRunnable.run(Native Method)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:942)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(SourceFile:1)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:240)
                                                                                                    	at android.os.Looper.loop(Looper.java:351)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(SourceFile:38)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
2024-05-15 17:45:25.600  3796-8276  GmsConnectionFactory    com.google.android.gms.persistent    E  UID for network attribution might be misset for tag 805307399 [CONTEXT service_id=39 ]
                                                                                                    elkx: FULL

**Comment by asurare** [comment]
Currently having the same issue

**Comment by velrajpaulraj** [comment]
> Currently having the same issue

Yes facing the same issue.

**Comment by asurare** [comment]
Only working using these options for now :
```jsx
FileLogger.configure({ maximumNumberOfFiles: 1, dailyRolling: false, maximumFileSize: 0 })
```

**Comment by velrajpaulraj** [comment]
> Only working using these options for now :
> 
> ```js
> FileLogger.configure({ maximumNumberOfFiles: 1, dailyRolling: false, maximumFileSize: 0 })
> ```

Thank you will try this solution

**Comment by velrajpaulraj** [comment]
> > Only working using these options for now :
> > ```js
> > FileLogger.configure({ maximumNumberOfFiles: 1, dailyRolling: false, maximumFileSize: 0 })
> > ```
> 
> Thank you will try this solution

@asurare  Still facing the same issue only on release build.

**Comment by fdrault** [comment]
Hello guys, thank you for reporting this.

I don't have time right now, but I might take a look next week

Can you provide the following information to make my work easier:
- Exact react-native version
- platform (iOS/Android/both)
- Expo or Bare
- What your `FileLogger.configure()` looks like.

**Comment by makalaravikiran** [comment]
Same issue for me as well

RN Version: 0.74.1
Platform: Android
Bare react native
new Arch enabled
FileLogger.configure({captureConsole:true})

**Comment by fdrault** [comment]
I tried to reproduce your issue on an Android 14 (Pixel 6) and Android 10, in release mode, without any success.

Do you use the last version of our library, v0.5.5 ? 
If not, please try to update using `npm install react-native-file-logger@latest` or `yarn add  react-native-file-logger`.

On a pure new application, I am unable to reproduce this issue...

**Comment by vinterstum** [comment]
Same, after upgrade to RN 0.73.4 and a bunch of other Android SDK updates that were required for that. Crashes no matter which options I pass to FileLogger.configure.

react-native 0.73.4
react-native-file-logger 0.5.5

        buildToolsVersion = "34.0.0"
        minSdkVersion = 23
        compileSdkVersion = 34
        targetSdkVersion = 33
        ndkVersion = "26.1.10909125"
        kotlinVersion = "1.7.10"

**Comment by vinterstum** [comment]
If I turned off dailyRolling:  Fatal Exception: java.lang.IllegalStateException: FileNamePattern [/storage/emulated/0/Android/data/no.symi.seyfr.android.prod/cache/logs/no.symi.seyfr.android.prod-%i.log] does not contain a valid IntegerToken 

Doesn't make a whole lot of sense, unless these strings are suddenly being escaped somehow.

**Comment by wzdxy** [comment]
Same issue
```
"react-native": "0.73.2" 
"react-native-file-logger": "^0.5.5"
"expo": "50.0.18"
```
My test device is PCT-AL10 with Android 10
The error will occur **only  in release mode**, there is no problem in dev mode.

```ts
// my code
FileLogger.configure({
    formatter: (level, msg) =>
      `${dayjs().format('YYYY-MM-DD HH:mm:ss')} [${LogLevel[level]}] ${msg}`,
    maximumNumberOfFiles: 30,
  }).then((res) => {
    console.log('FileLogger configure success', res)
  })
```

Below are the logs collected by Sentry：
<img width="812" alt="image" src="https://github.com/user-attachments/assets/2670095c-d9b1-4c98-b3e6-a1ba9818fb7e">
<img width="807" alt="image" src="https://github.com/user-attachments/assets/2757dd58-e30c-4265-b83b-46515ed36e03">

**Comment by ngoclamnn** [comment]
getting the same issue with android release build after trying to upgrade to react-native 0.74.3

**Comment by ngoclamnn** [comment]
there is a fix in this repo https://github.com/tony19/logback-android/issues/344
adding `-keepclassmembers class ch.qos.logback.core.rolling.helper.* { <init>(); }` to the android/app/proguard-rules.pro helped

**Comment by fdrault** [comment]
Thank you all for information around this issue.
@ngoclamnn nicely pointed out the exact root of the problem  

The issue occurs only when Proguard is enabled, that is why it occurs only on Release mode with Proguard, but not with the basic expo/react-native-cli init project.

🚀 Fix published in version `0.5.6`

Please update the library:
```
npm install react-native-file-logger@latest
```
